cns/restserver/api.go

```diff
@@ -1139,7 +1139,7 @@ func (h *HTTPRestService) doPublish(ctx context.Context, req cns.PublishNetworkC
 	var innerReq nmagent.PutNetworkContainerRequest
 	err := json.Unmarshal(innerReqBytes, &innerReq)
 	if err != nil {
-		returnMessage := fmt.Sprintf("Failed to publish Network Container: %s", req.NetworkContainerID)
+		returnMessage := fmt.Sprintf("Failed to publish Network Container in unmarshalling request: %s with err: %v", req.NetworkContainerID, err)
```

Comment:
Can we improve the message here? Reads a bit odd IMO. What failed was the unmarshaling, not the publishing.
```suggestion
		returnMessage := fmt.Sprintf("Failed to unmarshal embedded NC publish request for NC %s, with err: %v", req.NetworkContainerID, err)
```
or something along those lines.